### ISSUE TITLE
Allow a custom (modifier) class for multiselect

### ISSUE THREAD

**Issue by FreekVR** [shentao/vue-multiselect#667]
Hi! I love how multiselect is built using BEM, which nicely matches with our workflow.

However we have several variations of multiselect on our site, and we already have two different styles for these dropdowns. The only possibility I have now is to add styling based on inheritance (eg wrap it in a div), which is kind of anti-BEM ;)

I would prefer to be able to pass a custom prop or option that allows for an additional modifier, so that the base class "multiselect" would become "multiselect multiselect--filter" for example.

**Comment by shentao** [comment]
Given the current styles are static, changing the base class from `multiselect` to something different would probably break the whole component.

What kind of changes do you need? Maybe those can be accomplished by using some of the slots?

**Comment by FreekVR** [comment]
I wouldn't suggest replacing the class (that would break things for us too), but being able to add a custom additional class as modifier.

**Comment by shentao** [comment]
Yeah, I think it should be possible to add something like a `class-name` prop that would accept another class. This should work with BEM-like classes as well as CSS Modules. Scoped CSS would probably need to use the deep `>>>` operator.

Feel free to submit a PR. The release will be done once I finish the refactor I started on master (current tests are failing).

**Comment by lijiahao8898** [comment]
I have the same problem.  >.<!

**Comment by otijhuis** [comment]
Maybe I don't understand the exact issue but isn't this already possible with how Vue behaves?

Attributes on the multiselect tag will automatically be inherited and merged with the attributes of the top element of the component. Which in this case is the div that has the multiselect class. If you add `class="multiselect--filter"` to the multiselect tag the div should get both the *multiselect* and *multiselect--filter* classes.

**Comment by dlindahl** [comment]
In my current research, `class` is automatically inherited and merged if you are using templates. However, if you are using a render function, that does not appear to be the case. Multiselect would need to explicitly add `:class="class"` in order to pull it in.

I could be wrong, but thats what I am seeing so far.
